src/bom/bitcode.rs

```diff
@@ -1124,6 +1142,43 @@ mod tests {
         }
     }
 
+    fn clean_temp_in_args(args: &Vec<OsString>) -> Vec<OsString> {
+        // The tests here would like to compare the actual args against a
+        // known set, but sometimes the actual args contain generated
+        // temporary files.  This function applies various heuristics to
+        // identify and strip or convert those temporary files into a static
+        // pattern for the simple equality comparisons below.  These
+        // heuristics are based on the patterns used for those temp files in
+        // the main code.
+        args.iter().map(clean_temp_in_arg).collect()
+    }
+    fn clean_temp_with_suffix(argstr: &String, sfx: &str) -> String {
+        if argstr.ends_with(sfx) {
+            // If the argstr is something like "--foo=/tmp/dir/is/tmpfile{sfx}",
+            // tries to just remove the file portion and return "--foo={sfx}".
+            match argstr.find("/") {
+                Some(i) => {
+                    let mut r = String::new();
+                    r.push_str(&argstr[..i]);
+                    r.push_str(sfx);
+                    r
+                },
+                None => String::from(sfx)
+            }
+        } else {
+            String::from(argstr)
+        }
+    }
+    fn clean_temp_in_arg(arg: &OsString) -> OsString {
+        let argstr = arg.to_string_lossy();
+        clean_temp_with_suffix(
+            &clean_temp_with_suffix(&String::from(argstr),
+                                    "discard{section-file}"),
+            "discard{out-file}")
```

Comment:
It could be a separate PR, but we should really make these shared constants to avoid typo bugs later

Reply:
The only duplication of these is in the tests, and typos will cause the tests to fail.  I had avoided lazy_static common definitions due to complexity, but I can add those if you prefer that approach.

Reply:
I would like to defer that to a separate PR though, if that's OK.

Reply:
That is fine. Also they don't need to be lazy statics. Since they are just string constants, they can be:

```
const OUT_FILE_NAME : &'static str = "discard{out-file}";
```

Reply:
That `&` makes all the difference.  I had tried it as a `const str` and got errors, which I thought I'd need to resort to `lazy_static` to solve.